src/AzureEventGridSimulator/Program.cs

```diff
@@ -14,46 +14,53 @@ namespace AzureEventGridSimulator
 {
     public static class Program
     {
-        public static void Main()
+        public static void Main(string[] args)
         {
             try
             {
-                var host = WebHost.CreateDefaultBuilder()
-                                  .ConfigureAppConfiguration((context, builder) =>
-                                  {
-                                      var env = context.HostingEnvironment;
+                var host = WebHost
+                    .CreateDefaultBuilder(args)
+                    .ConfigureAppConfiguration((context, builder) =>
+                    {
+                        var env = context.HostingEnvironment;
 
-                                      var config = builder.AddJsonFile("appsettings.json", false, false)
-                                                          .AddJsonFile($"appsettings.{env.EnvironmentName.ToLowerInvariant().Trim()}.json", true, false)
-                                                          .AddEnvironmentVariables()
-                                                          .Build();
+                        var config = builder.Build();
 
-                                      Log.Logger = new LoggerConfiguration()
-                                                   .Enrich.FromLogContext()
-                                                   .Enrich.WithProperty("AspNetCoreEnvironment", env.EnvironmentName)
-                                                   .Enrich.WithMachineName()
-                                                   .MinimumLevel.Debug()
-                                                   .MinimumLevel.Override("Microsoft", LogEventLevel.Error)
-                                                   .MinimumLevel.Override("System", LogEventLevel.Error)
-                                                   .ReadFrom.Configuration(config)
-                                                   .CreateLogger();
-                                  })
-                                  .UseStartup<Startup>()
-                                  .UseSerilog()
-                                  .UseKestrel(options =>
-                                  {
-                                      var simulatorSettings = (SimulatorSettings)options.ApplicationServices.GetService(typeof(SimulatorSettings));
+                        var configFile = config.GetValue<string>("ConfigFile");
+                        if (!string.IsNullOrEmpty(configFile))
+                        {
+                            builder.AddJsonFile(configFile, optional: false);
```

Comment:
If you specify the `ConfigFile` setting we add that as a config source as not optional.

We *could* say that if you don't specify that option we add `appsettings.json` as not optional?

Reply:
Yeah that sounds reasonable.

Reply:
I've not even tried running it without a appsettings.json file present.